### DISCUSSION TITLE
Textures not binded when loading model

### DISCUSSION THREAD

**Discussion by m-hilani** [MuhammadSulaiman001/opengl-lab#29] [Q&A]
hello Mister, I am trying to import a model to the same LearnOpenGl model loading project  but the model  is in black  .

**Comment by MuhammadSulaiman001** [comment]
Hi @m-hilani. Please provide a minimal code to reproduce the issue (upload a sample project to your GitHub repo if you want).

**Reply by m-hilani** [reply]
![Screenshot 2023-12-26 223158](https://github.com/MuhammadSulaiman001/opengl-lab/assets/148714322/1e42d3f6-aa59-4c2b-8222-d24447cef0b0)
![Screenshot 2023-12-26 223216](https://github.com/MuhammadSulaiman001/opengl-lab/assets/148714322/2b69b131-85c3-4d6d-8940-9a0c7d97457c)
![Screenshot 2023-12-26 222929](https://github.com/MuhammadSulaiman001/opengl-lab/assets/148714322/b993c1fe-4f14-4be3-90f7-f680e606df6a)

actually I only have edited the code but it won't work

**Reply by MuhammadSulaiman001** [reply]
There should be an `.mtl` file like this [backpack.mtl](https://github.com/JoeyDeVries/LearnOpenGL/blob/master/resources/objects/backpack/backpack.mtl), try open your .mtl file in any text editor (ex. Notepad++), and check that the paths to the textures are correct.

![Screenshot 2023-12-27 101907](https://github.com/MuhammadSulaiman001/opengl-lab/assets/36730723/87f3e987-cadd-45e0-ac99-b6ec1c454fff)
[Source](https://learnopengl.com/Model-Loading/Model)


If this is not the case, I would ask for a minimal demo project so we can solve the issue more quick.

**Reply by m-hilani** [reply]
sorry for being late https://github.com/m-hilani/AlAqsa.git  i added  it to mybranch branch

**Reply by m-hilani** [reply]
@MuhammadSulaiman001 I made a new branch in visual but i didn't see it . so i pushed it using cmd then i saw it pushed from 6 hours ?

**Reply by m-hilani** [reply]
![Screenshot 2023-12-27 213305](https://github.com/MuhammadSulaiman001/opengl-lab/assets/148714322/362387c6-6424-463a-966a-b1c09aec0f07)
that's my last progress the textures are binded but not in proper way and it becomes very slow unfortunately
 although i view it on https://www.creators3d.com/online-viewer so i think the problem is not in the model

**Reply by MuhammadSulaiman001** [reply]
Hi @m-hilani, I've checked your fork, here are what I found:
1. Kotel wall is loaded correctly, here is the output on my machine (the model is very large, > 30 MB, so I commented out the code for loading the second model)
![Screenshot 2023-12-28 200435](https://github.com/MuhammadSulaiman001/opengl-lab/assets/36730723/34d08a97-c3e6-47d2-a680-9baf648c3542)
2. I've used [3D Viewer](https://apps.microsoft.com/detail/9NBLGGH42THS?hl=en-us&gl=US) from Microsoft, it could not load the rest of the models, I suspect that these models are invalid.
![image](https://github.com/MuhammadSulaiman001/opengl-lab/assets/36730723/8399d194-b76f-48a5-9f5c-514a0e623aff)

**Reply by MuhammadSulaiman001** [reply]
Also, Models of this size will make the application slow for sure! The size of the model is very critical setting, try using small models ( > 5 MB).

Good luck.

**Reply by MuhammadSulaiman001** [reply]
1. I see, I got your point, the textures are bound incorrectly in LearnOpenGL demo, as opposite to the output at 3D Viewer. Definatily an issue with the loader class [model.h](https://github.com/JoeyDeVries/LearnOpenGL/blob/master/includes/learnopengl/model.h).
2. I think it's wrong to use a 3d model for a adding a wall to the scene! Why not just using a texture in this case?! much faster, good looking and saves computer resources,
3. Good luck again.

**Reply by m-hilani** [reply]
Thanks
